### DISCUSSION TITLE
Attribute not found while testing a Beanie Model inside fast api

### DISCUSSION THREAD

**Discussion by pythonhubdev** [fastapi/fastapi#6321] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [x] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
My Code:


My Route:

@router.post("/login")
async def internalLogin(request: Request,
                        email: str = Form(...),
                        password: str = Form(...)):
    try:
        res, token = await Controller.internalLogin(email=email, password=password)
        if res:
            return {"message": "Success"}
        else:
           return {"message": "Failure"}
    except DocumentNotFound as documentNotFoundException:
        return {"message": "Error"}
```

Controller:
```
@staticmethod
    async def internalLogin(email: str, password: str) -> List[bool | str]:
        logger.info(message="Inside OpenApi Controller", fileName=__name__, functionName="OpenApiController")
        try:
            user = await internalUserDb(email=email)
            if user is not None and user.verifyPassword(password):
                print("Logged In")
                return [True, ""]
            else:
                print("Failed)
                return [False, ""]
        except DocumentNotFound as documentNotFound:
            raise documentNotFound

```

DB:

```
async def internalUserDb(email: str) -> InternalUserModel:
    try:
        user: InternalUserModel = await InternalUserModel.find_one(InternalUserModel.email == email)
        return user
    except DocumentNotFound as documentNotFound:
        raise documentNotFound
```

My TestCode:

```
@pytest.mark.anyio
async def testLogin():
    response = await asyncClient.post("/internalLogin",
                                      data={"email": "sample@mail.com", "password": "samplePass"})
    assert response.status_code == 303
```

My error while testing is: 

```
FAILED Tests/TestLogin.py::testLogin[asyncio] - AttributeError: type object 'InternalUserModel' has no attribute 'email'
FAILED Tests/TestLogin.py::testLogin[trio] - AttributeError: type object 'InternalUserModel' has no attribute 'email'
```

### Description

Hello, I am new to FastAPI. I am trying to test the fast api with PyTest. Normal tests are working perfectly fine but I am using MongoDB as backend to store my data. While I try to test the route that does some data fetching from database it shows error like `attribute not inside the model`. I am using Beanie ODM for MongoDB.

### Operating System

macOS

### FastAPI Version

0.73

### Python Version

3.10

**Comment by odiseo0** [comment]
Can you play show the code of `InternalUserModel`?
Also, do you pass the Beanie object to a pydantic model or you directly use Beanie?

**Comment by pythonhubdev** [comment]
```
from beanie import Document

class InternalUserModel(Document):
    email: str
    password: str
    
    class Collection:
        name = "Internal Users"

    class Config:
        arbitrary_types_allowed = True
```

**Comment by pythonhubdev** [comment]
> Can you play show the code of `InternalUserModel`? Also, do you pass the Beanie object to a pydantic model or you directly use Beanie?

I directly use beanie

**Comment by pythonhubdev** [comment]
Any update on this please?

**Comment by agustin-del-pino** [comment]
Hello,
I've the same issue, is there some fix or workaround about this?.

**Comment by TheRealLorenz** [comment]
Hi there, in my case the problem was caused by the @app.on_event("startup) decorator which didn't execute my init function. I solved it by wrapping the TestClient instance with a `with` statement, like this:

```python
with TestClient(app) as client:
    # do stuff
```

For further information: https://fastapi.tiangolo.com/advanced/testing-events/

**Comment by d2davidtb** [comment, marked as answer]
Hi, in my case, I had forgot to add model on init_beanie. 😂 

```
init_beanie(
  database=client.get_default_database(),
  document_models=[
    InternalUserModel,
    ...
  ]
)
```
